subprojects/gradle/namespaced-resources-fixer/src/main/kotlin/com/avito/android/plugin/FileResourcesFixer.kt

```diff
@@ -0,0 +1,169 @@
+package com.avito.android.plugin
+
+import io.github.detekt.parser.KtCompiler
+import org.jetbrains.kotlin.com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.com.intellij.psi.PsiWhiteSpace
+import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtImportDirective
+import org.jetbrains.kotlin.psi.KtImportList
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.psi.KtTreeVisitorVoid
+import java.io.File
+
+/**
+ * Replace unambiguous R class references to Application by libraries R classes
+ */
+internal class FileResourcesFixer {
+
+    private val compiler = KtCompiler()
+
+    /**
+     * @return new file content or null if nothing changed
+     */
+    fun fixMergedResources(file: File, app: AndroidModule, libraries: List<AndroidModule>): String? {
+        val ktFile = compiler.compile(file.toPath(), file.toPath())
```

Comment:
Do you mean a classFile? Or compiler is a parser from text to kotlinPsi?

Reply:
It's a `org.jetbrains.kotlin.psi.KtFile`.
As far as I understand, it's a lightweight AST, not compiled.

Reply:
moved to `parse` extension function to reflect this intention